docs/topics/jvm/java-interop.md

```diff
@@ -153,6 +154,29 @@ When the `@NotNull` annotation is missing from a type argument, you get a platfo
 fun toSet(elements: (Mutable)Collection<String!>) : (Mutable)Set<String!> { ... }
 ```
 
+Kotlin also takes into account nullability annotations on type arguments of base classes and interfaces. For example, 
+there are 2 Java classes with the signatures provided below.
+
+```java
+public class Base<T> { ... }
+```
+
+```java
+public class Derived extends Base<@Nullable String> { ... }
+```
+
+In the Kotlin code, passing the instance of `Derived` where the `Base<String>` assumed produces the warning.
+
+```kotlin
+fun takeBaseOfNotNullStrings(x: Base<String>) {}
+
+fun main() {
+    takeBaseOfNotNullStrings(Derived()) // warning: nullability mismatch
+}
+```
+
+The upper bound of `Derived` is set to `Base<String?>`, which is not a subtype of `Base<String>`.
```

Comment:
@p7nov In the previous release, you had a concern about the warning description. Is it now clear for a reader what is the warning and why does it appear?

Reply:
Yes, this is good. However, variance is an advanced subject, so a link would be helpful. I'd add:
> Learn more about the [variance of generic classes](https://kotlinlang.org/docs/generics.html#variance).